helm-chart/PF_RING.md

```diff
@@ -0,0 +1,183 @@
+
+Example output for a compatible cluster:
+```
+Node                                          Kernel Version                 Supported
+ip-192-168-77-230.us-west-2.compute.internal  5.10.199-190.747.amzn2.x86_64  true
+ip-192-168-34-216.us-west-2.compute.internal  5.10.199-190.747.amzn2.x86_64  true
+
+Cluster is compatible
+```
+
+
+### Pre-built kernel module exists and external egress allowed
+
+If PF_RING kernel modules are already available for the target nodes (cluster is compatible), both `auto` and `kmm` modes are applicable.
+
+|auto|kmm|
+|----|---|
+| `SYS_MODULE` capability required for Kubeshark | `SYS_MODULE` capability is not required for Kubeshark|
+| no additional dependencies | (!)requires `cert-manager` and `KMM` installed (follow [instructions](https://kmm.sigs.k8s.io/documentation/install/)) |
+| Kubeshark falls back to `libpcap` if `PF_RING` kernel module not available | Kubshark waits until PF_RING is loaded with KMM|
+| module is downloaded from S3 bucket in AWS | module is loaded from `kubeshark/pf-ring-module:<kernel version>` container|
+| requires egress connectivity to AWS S3 endpoints | requires egress connectivity to Docker Hub container registry|
+
+
+### Pre-built kernel module doesn't exist or external egress isn't allowed
+
+In cases where PF_RING kernel modules are not yet available for the target nodes, or if external egress is restricted, the `kmm` mode is the only viable option (`auto` mode would start Kubeshark with libpcap, not PF_RING).
+This approach enables the use of custom container images as the source for PF_RING kernel modules and allows leveraging private container registries.
+
+#### Steps to Use kmm with Custom Containers
+
+1. Compile the pf_ring.ko kernel module for your target kernel version (see [Appendix B](#appendix-b-pf_ring-kernel-module-compilation) for instructions).
+
+After building the module with kubeshark pfring compile, you will obtain a `pf-ring-<kernel version>.ko` file.
+If manually built, rename the kernel module to this format.
+
+2. Build and push Docker container(-s) with the kernel module file from stage 1.
+
+Create `Dockerfile` in the folder with PF_RING kernel module:
+
+```
+FROM alpine:3.18
+ARG KERNEL_VERSION
+
+COPY pf-ring-${KERNEL_VERSION}.ko /opt/lib/modules/${KERNEL_VERSION}/pf_ring.ko
+RUN apk add kmod
+
+RUN depmod -b /opt ${KERNEL_VERSION}
+```
+
+Run build&command:
+
+```
+docker build --build-arg <kernel version> <your registry>/<image>:<kernel version>
+docker push <your registry>:/<image>:<kernel version>
+```
+
+It is recommended to use kernel version as a container tag for consistency.
+
+
+3. Configure Helm values
+
+```
+tap:
+  kernelModule:
+    mode: kmm
+    kernelMappings:
+    - regexp: '<kernel version>'
+      containerImage: '<your-registry>/<image>:<kernel version>'
+    imageRepoSecret: <optional secret with credentials for private registry>
```

Comment:
Yeah, I saw it.
But frankly speaking, I'd keep it explicit. Because it is done now, kernelMappings and imageRepoSecret directly map to KMM Module CR (https://kmm.sigs.k8s.io/documentation/deploy_kmod/) fields and are required only in scope of pf_ring. 
If we mix `tap.docker` with `kernelModule` configurations, it start to look a bit messy IMO. E.g., what if 
1. someone uses kubeshark docker repo to install kubeshark,
2. inds pf_ring is not available, builds its own pf_ring container in private registry
3. now needs to configure secret to pull from private registry
4. ends up having `tap.docker.imagePullSecrets` configured (which in fact are not used for `tap.docker.registry`, but for `tap.kernelModule.kernelMappings`)

that looks confusing